src/FSLibrary/StellarNetworkData.fs

```diff
@@ -22,36 +22,117 @@ let TestnetLatestHistoryArchiveState =
 type PubnetNode = JsonProvider<"json-type-samples/sample-network-data.json", SampleIsList=false, ResolutionFolder=cwd>
 type Tier1PublicKey = JsonProvider<"json-type-samples/sample-keys.json", SampleIsList=false, ResolutionFolder=cwd>
 
+// When scaling the network, we need to pick
+// the degree of each new node.
+// The following numbers are added here based on pubnet observations/educated guesses.
+let peerCountTier1 (random: System.Random) : int = random.Next(25, 81)
+
+let peerCountNonTier1 (random: System.Random) : int = if random.Next(2) = 0 then 8 else random.Next(1, 71)
+
+// For simplicity, we will make each new tier-1 organization contain exactly 3 nodes
+// while scaling the pubnet.
+let tier1OrgSize = 3
+
+let extractEdges (graph: PubnetNode.Root array) : (string * string) array =
+    let getEdgesFromNode (node: PubnetNode.Root) : (string * string) array =
+        node.Peers
+        |> Array.filter (fun peer -> peer < node.PublicKey) // This filter ensures that we add each edge exactly once.
+        |> Array.map (fun peer -> (peer, node.PublicKey))
+
+    graph |> Array.map getEdgesFromNode |> Array.reduce Array.append
+
+// Given `edgeSet` containing edges (each edge is represented as a pair of strings),
+// creae a map whose key is a pubkey and the corresponding value is the list of
+// nodes it's connected to.
+let createAdjacencyMap (edgeSet: Set<string * string>) : Map<string, string list> =
+    let edgeList : (string * string) list = Set.toList edgeSet
+
+    let adjacencyList : (string * (string list)) list =
+        edgeList
+        |> List.append (List.map (fun (x, y) -> (y, x)) edgeList)
```

Comment:
not sure I follow this function. Why do you need to swap vertices?

Reply:
I added comments to the function. The code keeps exactly one of `(a, b)` or `(b, a)` as the graph is undirected (and also it makes it easier to track edges). But, I needed to add both `(a, b)` and `(b, a)` to create an adjacency list.